### ISSUE TITLE
web interface

### ISSUE THREAD

**Issue by k2m30** [Motion-Project/motion#381]
There is no possibility to customize UI for the Motion web interface.
I would suggest adding classes and ids for html tags inside `webhttpd.c` and `default.css` file for further customizing.

Ready to make a pull request if it is acceptable.

Something like http://crankycoder.net/wp-content/uploads/2016/06/BeQZIS7CEAAemgw.png UI would be much nicer. I would take generic Twitter Bootstrap as a beginning.

**Comment by tosiara** [comment]
You have provided a screenshot of motioneye as an example of good web ui. So then what stops you from using motioneye?

**Comment by jogu** [comment]
The motion web interface isn't really intended to be a UI IMHO.

An alternative approach would be to improve what is there to be a proper REST API so that people who want to write UIs can easily do so.

**Comment by iskunk** [comment]
What @jogu said. The interface is deliberately minimal because making a full-fledged modern Web interface is a whole major project unto itself... and C code is about the worst place to do it from. Motion is meant to be more the backend piece to a shiny frontend like MotionEye.

Adding CSS classes/IDs is a nice idea in itself (presumably combined with a parameter specifying an external stylesheet). But the Web UI already is in need of an overhaul---there's been talk of replacing it with libmicrohttpd or a similar modern embedded Web server. Work that extends `webhttpd.c` rather than replacing or bug-fixing it would, to a degree, be wasted effort.

**Comment by k2m30** [comment]
@tosiara motion itself is very light and simple. But motioneye requires python, all those libs, dependencies and so on. Which is quite complicated for me. At the same time, some relatively easy css makeup (like a day or two of work for me) would make a great difference. 

However, if it is against current roadmap, it is ok.

**Comment by iskunk** [comment]
I've always thought it should be possible to build a really snazzy AJAX-ish UI to Motion that is entirely client-side (or at least where the server side is minimal, unlike MotionEye). I could certainly see us working with someone who is building such a thing, to ensure they have all the hooks they need.

A lot of work needs to be done on this side first, however. The program architecture is quite old and doing  any sort of cool new feature is a lot harder than it needs to be.

**Comment by tosiara** [comment]
Adding few css and js load, as well as class anchors wouldn't really hurt and should not be a big change. @k2m30, if you volunteer to create proof of concept pull - you are welcome, let's see how it goes?

**Comment by jogu** [comment]
I should probably clarify as my earlier statement might have implied otherwise: As tosiara says, a pull request that neatens up the web interface would be very welcome.

**Comment by k2m30** [comment]
@tosiara @jogu Ok, I'll do it

**Comment by zucodaf** [comment]
Could you add or make the Screenshoots and Streaming Videos look like polaroid photo frames for still photos and like old Dropcam interface for videos. These would look so awesome. Thank you!

https://www.pubnub.com/wp-content/uploads/2015/06/screen2.gif
https://github.com/pubnub/SecurityCam

https://itp.nyu.edu/classes/cdni-spring2014/files/2014/03/dropcam-visitorview.png

**Comment by iskunk** [comment]
Polaroid photos? Someone's showing their age  ^_^

(It's not a bad idea, just a visual element that is becoming less and less obvious as time goes on.)

That DropCam interface is pretty, but a timeline UI like that would probably belong more to a MotionEye-type project.

**Comment by telemaxx** [comment]
with Ticket #437 i have postet a feature request.
The text inside the UI is sooo small on mobile devices.
For that reason apple has introduced the
[Viewport Metatag](https://developer.apple.com/library/content/documentation/AppleApplications/Reference/SafariWebContent/UsingtheViewport/UsingtheViewport.html)

With that the text looks really good.

This is also working on other Browsers.

[Mozilla](https://developer.mozilla.org/en-US/docs/Mozilla/Mobile/Viewport_meta_tag)
[w3.org](https://www.w3.org/TR/mwabp/#bp-viewport)

Only one line must be addet in webhttpd.c

see #437

**Comment by Harvie** [comment]
Well i think that motion should decide wheter it wants to provide real UI or not.
I think what @jogu says makes lots of sense. Strip motion to only have basic REST API and move all other functionality to semi-separate project like motioneye. However you should adopt such program and make it part of motion itself (while being separate code), so perfect compatibility and project coordination can be achieved. That would allow all efforts to be going same way. Currently there are lots of motion web ui projects, but none of them is perfect, because they are developed somewhat separately from motion itself.

**Comment by Mr-Dave** [comment]
With the current roadmap focused on limited enhancements,  existing functionality stability and code clean up, I would be reluctant to adopt another project to take on web UI functionality.  And until such a project is adopted, it would seem to be a step backward to remove the limited functionality that the web interface currently provides.  

To address the original issue posted, a full UI is not in my current plan.  It would only be code clean up for the webhttpd.c to make it easier to maintain.  If security or small revisions are needed/desired to the existing interface, pull requests can be provided.

I will leave this issue open for a couple of more days for comments and then be closing it.

**Comment by Mr-Dave** [comment]
For those reviewing the old issues, the current master now includes a revised web user interface as a result of the code clean up activities.

**Comment by Harvie** [comment]
@Mr-Dave Nice! Do you have any screenshots?

**Comment by Mr-DaveDev** [comment]
I don't have an account anywhere to post pics so here are some small ones you can zoom in on.
English variations first then are images using foreign languages(not validated by native speakers).  
See (and contribute) on issue #440 for translations.  There are also still some outstanding issues on the interface that are being tracked on #661 and we need to find some javascript/css expertise.  For future comments, please use those issues.

![image](https://user-images.githubusercontent.com/21115374/36884435-43084c40-1d9e-11e8-8a70-b456df06727c.png)
Foreign (Translations also permitted in drop down next to option see description setup_mode)
![image](https://user-images.githubusercontent.com/21115374/36884524-de312d68-1d9e-11e8-9e36-9a379c930e3d.png)

**Comment by josvanr** [comment]
Looking for a way to customize the webinterface I came upon this thread. The interface doesnt allow this but I found a different solution. Will post it in case anyone needs this. 

It can be done using chrome devtools: one can make modifications in a running website. I wanted to display 2 cams with the entire width of the  browser filled. So I just open devtools while looking at the webinterface, and changed 25% width to 49%, then the window looks like in the screenshot. Changes can be made persistent by using 'overrides' in devtools. To view the modified interface after reloading, one has to open and close devtools again (shift-ctrl-j), though. It works for me...


![1](https://user-images.githubusercontent.com/35400537/96336250-babffc80-107e-11eb-9188-2e24cd407562.jpg)
